saleor/graphql/account/mutations/staff.py

```diff
@@ -0,0 +1,379 @@
+    def clean_input(cls, info, instance, data):
+        cleaned_input = super().clean_input(info, instance, data)
+
+        # set is_staff to True to create a staff user
+        cleaned_input["is_staff"] = True
+
+        # clean and prepare permissions
+        if "permissions" in cleaned_input:
+            permissions = cleaned_input.pop("permissions")
+            cleaned_input["user_permissions"] = get_permissions(permissions)
+        return cleaned_input
+
+    @classmethod
+    def save(cls, info, user, cleaned_input):
+        create_avatar = not user.avatar
+        if create_avatar:
+            user.avatar = get_random_avatar()
+        user.save()
+        if create_avatar:
+            create_user_avatar_thumbnails.delay(user_id=user.pk)
+        if cleaned_input.get("send_password_email"):
+            send_set_password_staff_email.delay(user.pk)
+
+
+class StaffUpdate(StaffCreate):
+    class Arguments:
+        id = graphene.ID(description="ID of a staff user to update.", required=True)
+        input = StaffInput(
+            description="Fields required to update a staff user.", required=True
+        )
+
+    class Meta:
+        description = "Updates an existing staff user."
+        exclude = ["password"]
+        model = models.User
+        permissions = ("account.manage_staff",)
+
+    @classmethod
+    def clean_is_active(cls, is_active, instance, user):
+        if not is_active:
+            if user == instance:
+                raise ValidationError(
+                    {"is_active": "Cannot deactivate your own account."}
+                )
+            elif instance.is_superuser:
+                raise ValidationError(
+                    {"is_active": "Cannot deactivate superuser's account."}
+                )
+
+    @classmethod
+    def clean_input(cls, info, instance, data):
+        cleaned_input = super().clean_input(info, instance, data)
+        is_active = cleaned_input.get("is_active")
+        if is_active is not None:
+            cls.clean_is_active(is_active, instance, info.context.user)
+        return cleaned_input
+
+
+class StaffDelete(StaffDeleteMixin, UserDelete):
+    class Meta:
+        description = "Deletes a staff user."
+        model = models.User
+        permissions = ("account.manage_staff",)
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a staff user to delete.")
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        if not cls.check_permissions(info.context.user):
+            raise PermissionDenied()
+
+        user_id = data.get("id")
+        instance = cls.get_node_or_error(info, user_id, only_type=User)
+        cls.clean_instance(info, instance)
+
+        db_id = instance.id
+        remove_staff_member(instance)
+        # After the instance is deleted, set its ID to the original database's
+        # ID so that the success response contains ID of the deleted object.
+        instance.id = db_id
```

Comment:
We don't delete the user here, so we don't need this

Reply:
We delete the user if they have no orders placed.